src/core/serialize/src/main/scala/params/ByteArrayParam.scala

```diff
@@ -12,4 +14,15 @@ class ByteArrayParam(parent: Params, name: String, doc: String, isValid: Array[B
     def this(parent: Params, name: String, doc: String) =
       this(parent, name, doc, ParamValidators.alwaysTrue)
 
-  }
+    /** Creates a param pair with a java.util.List of values (for Java and Python). */
+    def w(value: java.util.List[java.lang.Byte]): ParamPair[Array[Byte]] =
+      w(value.asScala.map(_.asInstanceOf[Byte]).toArray)
+
```

Comment:
removed the jsonEncode/Decode as they were not being used anyway to fix this specific issue